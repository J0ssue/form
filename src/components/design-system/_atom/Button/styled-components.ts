import styled from "@emotion/styled";
import { css } from "@emotion/react";
//@ts-ignore
import { buildColor } from "components/design-system/_static/colorPalette";

export const ButtonContainer = styled.button<{ buttonType: string }>`
  display: flex;
  align-items: center;
  border: none;
  font-size: 13px;
  text-transform: uppercase;
  justify-content: center;
  letter-spacing: 1px;
  transition: background-color 0.2s, color 0.2s;
  cursor: pointer;
  background-color: transparent;

  ${({ buttonType }) =>
    buttonType === "primary" &&
    css`
      color: ${buildColor("white", "900")};
      background-color: ${buildColor("red", "100")};
      &:hover {
        background-color: ${buildColor("red", "000")};
      }
    `}

  ${({ buttonType }) =>
    buttonType === "secondary" &&
    css`
      color: ${buildColor("black", "900")};
      border: 2px solid ${buildColor("black", "900")};
      background-color: ${buildColor("white", "900")};
      &:hover {
        color: ${buildColor("white", "900")};
        background-color: ${buildColor("black", "900")};
      }
    `}

  ${({ buttonType }) =>
    buttonType !== "clear"
      ? css`
          width: 160px;
          height: 48px;
        `
      : css`
          &:hover {
            color: ${buildColor("red", "100")};
          }
        `}

  svg {
    margin-left: 13px;
  }
`;
