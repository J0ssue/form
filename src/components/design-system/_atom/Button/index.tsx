import React from "react";
import { ButtonContainer } from "./styled-components";

type ButtonTypes = "clear" | "primary" | "secondary";

export interface Props {
  /**
   * label for button
   */
  label: string;
  /**
   *  button type "clear" | "primary" | "secondary"
   */
  buttonType: ButtonTypes;
  /**
   *  onClick handler
   */
  onClickHandler: () => void;
}

const Button = (props: Props) => {
  const { label, buttonType, onClickHandler } = props;
  return (
    <ButtonContainer buttonType={buttonType} onClick={onClickHandler}>
      {label}
    </ButtonContainer>
  );
};

Button.defaultProps = {
  label: "button",
  buttonType: "clear",
};

export default Button;
