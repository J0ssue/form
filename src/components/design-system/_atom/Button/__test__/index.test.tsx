import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "..";
import userEvent from "@testing-library/user-event";

describe("<Button/> renders correctly", () => {
  const buttonClickHandler = jest.fn();

  it("should render <Button/> with default params", () => {
    const tree = renderer
      .create(<Button onClickHandler={buttonClickHandler} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render <Button/> with custom params", () => {
    const tree = renderer
      .create(
        <Button
          label="my button"
          buttonType="primary"
          onClickHandler={buttonClickHandler}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should run click handler when clicked", () => {
    render(<Button label="click button" onClickHandler={buttonClickHandler} />);

    userEvent.click(screen.getByText("click button"));

    expect(buttonClickHandler).toHaveBeenCalledTimes(1);
  });
});
