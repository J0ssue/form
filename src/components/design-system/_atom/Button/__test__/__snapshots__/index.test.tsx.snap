// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders correctly should render <Button/> with custom params 1`] = `
<button
  className="css-16327h2"
  onClick={[MockFunction]}
>
  my button
</button>
`;

exports[`<Button/> renders correctly should render <Button/> with default params 1`] = `
<button
  className="css-fx8zlh"
  onClick={[MockFunction]}
>
  button
</button>
`;
